modules/oe_editorial_corporate_workflow/modules/oe_editorial_corporate_workflow_translation/tests/Functional/CorporateWorkflowTranslationTest.php

```diff
@@ -0,0 +1,436 @@
+
+    // Create a new local translation task.
+    $this->drupalGet(Url::fromRoute('oe_translation.permission_translator.create_local_task', [
+      'entity' => $node->id(),
+      'source' => 'en',
+      'target' => 'fr',
+      'entity_type' => 'node',
+    ]));
+
+    // The default translation value comes from the previous version
+    // translation.
+    $this->assertSession()->elementContains('css', '#edit-title0value-translation', 'My node FR');
+
+    // Assert that the new job item got created and it has the revision ID of
+    // the validated node.
+    $job_items = $this->entityTypeManager->getStorage('tmgmt_job_item')->loadByProperties(['item_rid' => $validated_node->getRevisionId()]);
+    $this->assertCount(1, $job_items);
+
+    // Publish the node before finalizing the translation.
+    $this->moderateNode($node, 'published');
+    $revision_ids = $node_storage->revisionIds($node);
+    $this->assertCount(10, $revision_ids);
+
+    // Finalize the translation and check that the translation got saved onto
+    // the published version rather than the validated one where it actually
+    // got started.
+    $values = [
+      'title|0|value[translation]' => 'My node FR 2',
+    ];
+    // It should be the second local task item created so we use the ID 2.
+    $url = Url::fromRoute('entity.tmgmt_local_task_item.canonical', ['tmgmt_local_task_item' => 2]);
+    $this->drupalPostForm($url, $values, t('Save and complete translation'));
+    $node_storage->resetCache();
+    $validated_node = $node_storage->loadRevision($validated_node->getRevisionId());
+    // The second validated revision should have the old FR translation.
+    $this->assertEquals('My node FR', $validated_node->getTranslation('fr')->label());
+    $node = $node_storage->load($node->id());
+    // The new (current) published revision should have the new FR translation.
+    $this->assertEquals('My node FR 2', $node->getTranslation('fr')->label());
+
+    // The previous published revisions have the old FR translation.
+    $revision_ids = $node_storage->revisionIds($node);
+    /** @var \Drupal\node\NodeInterface[] $revisions */
+    $revisions = $node_storage->loadMultipleRevisions($revision_ids);
+    foreach ($revisions as $revision) {
+      if ($revision->isPublished() && (int) $revision->get('version')->major === 1) {
+        $this->assertEquals('My node FR', $revision->getTranslation('fr')->label());
+        break;
+      }
+    }
+  }
+
+  /**
+   * Tests that revision translations are carried over from latest revision.
+   *
+   * The test focuses on ensuring that when a new revision is created by the
+   * storage based on another one, the new one inherits the translated values
+   * from the one its based on and NOT from the latest default revision as core
+   * would have it.
+   *
+   * @see oe_editorial_corporate_workflow_translation_node_revision_create()
+   */
+  public function testTranslationRevisions(): void {
```

Comment:
Test was not firing oe_editorial_corporate_workflow_translation_node_presave().
After last commit 52672eb it is but test is failing.

Reply:
It's fixed.